### PR TITLE
Refine Select token

### PR DESCRIPTION
**Pull request by WIN-LIN**
# Description
- Based on #166 
- [Updated] Since we can send the token from both clicking the token in Home page and clicking the "Send" button, we need to managed the selected token across pages.
  - *Use url param to decide if the token is selected: `/send/:tokenAddress` (`0x` means not selected)
  - After integrating the Send flow, we could remove `TokenSendModal`.
- [Updated] Remove step state and use the required values to determine which step it is.
- Extract `tokenItem` and `tokenList` components

# Screenshots
### clicking the token in Home page

https://github.com/user-attachments/assets/2592d35a-ac9d-4b44-b890-b945554c893a

### clicking the "Send" button

https://github.com/user-attachments/assets/ab1e81bd-a70c-4017-8ca1-7fe55488d197

### CODE REVIEW

File: app/page/send/index.tsx [hunk 1]
```diff
@@ -1,50 +1,39 @@
...
+    <>
+      <StepBackHeader title="Select Token" />
+      <TokenList className="pt-[16px]">
+        {tokens.map((token, index) => (
+          <TokenItem
+            key={index}
+            token={token}
+            onClick={() => {
+              setTokenSelected(token)
+              navigate(`/send/${token.address}`)
```
Comment (pilagod) [inline app/page/send/index.tsx:27]: Q: Must we `navigate` at the end? Wouldn't `setTokenSelected` trigger re-render to next step for us?

File: app/path.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 export enum Path {
   Index = "/",
   Home = "/home",
-  Send = "/send",
+  Send = "/send/:tokenAddress",
```
Comment (pilagod) [inline app/path.ts:4]: Perhaps we can concat the variable in the send page? Adding it here would make it unavailable to use path for navigate like `navigate(Path.Send)`

File: app/util/getUserTokens.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { useAccount, useTokens } from "~app/hook/storage"
+import { getChainName } from "~packages/network/util"
+
+export const getUserTokens = () => {
```
Comment (pilagod) [inline app/util/getUserTokens.ts:4]: [future] I think maybe we could directly doing this in the useTokens storage hook. Could be considered in future PR.

File: app/component/tokenItem.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        <div className="flex-grow text-[20px] text-[#000000] text-left whitespace-nowrap">
+          {token.symbol}
+        </div>
+        <div className="flex flex-col items-end">
+          <div className="mb-[4px] text-[20px] font-[600] text-[#000000]">
+            {number.formatUnitsToFixed(token.balance, token.decimals, 2)}
+          </div>
+          {/* // TODO: add price oracle */}
+          <div className="text-[12px] text-[#000000]">$1.23</div>
+        </div>
```
Comment (pilagod) [inline app/component/tokenItem.tsx:26]: nit: I think we could just remove it at this moment

File: app/page/send/index.tsx (hunk 1 above)
Comment (WIN-LIN) [reply app/page/send/index.tsx]: [remove navigate when clicking TokenItem](https://github.com/pilagod/waallet-extension/pull/170/commits/a6a751bbf60dd3108c7d5e49d9ed2ae5f0dbe048)

File: app/util/getUserTokens.ts (hunk 3 above)
Comment (WIN-LIN) [reply app/util/getUserTokens.ts:4]: Sure! That will be more appropriate.

File: app/component/tokenItem.tsx (hunk 4 above)
Comment (WIN-LIN) [reply app/component/tokenItem.tsx]: [remove price oracle entry](https://github.com/pilagod/waallet-extension/pull/170/commits/d7dd4f1b1c4fdfc465456a56625dec67242ecfba)

File: app/path.ts (hunk 2 above)
Comment (WIN-LIN) [reply app/path.ts]: Sure! 
We still need to define the url schema in the `<PageRouter />`, or it cannot read `/send/${address}`.

[concate variable](https://github.com/pilagod/waallet-extension/pull/170/commits/09cc97da15fb9a282cce65c7023ac31b3960a782)

Comment (pilagod) [reply app/path.ts]: You're right. I saw so many places need the concat 🥲 I don't have good idea to organize them now, we can refine them when we get a good way.

**Review by pilagod — APPROVED** [review]

**Review by oneleo — APPROVED** [review]
Summary:
1. Apply `TokenList` div to `Home - Token` and `Send - Token` pages.
2. Use routing for passing and receiving `tokenAddress`.
3. Redirect `Token send` function from `Home - Token` to `Send - Token` page.
4. On `Send - Token` page:
(1) redirect to `SelectToken` if `tokenSelected` is missing, and
(2) to `SelectAddress` if `txTo` is missing.

**Pull request merged**